src/hip/hip_build_utils.cpp

```diff
@@ -39,6 +39,48 @@
 MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_HIP_VERBOSE)
 MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_HIP_DUMP)
 
+#if MIOPEN_OFFLINE_COMPILER_PATHS_V2
+
+// List to hold Relative Directory Path
+// for each Compiler Flags from ROCM Install Path
+// The definition of this variable expected to be available in C/C++
+// code of the Target Library.
+static const char* MIOPEN_CMP_FLAG_REL_PATHS[MIOPEN_CMP_MAX_Flag] = {
+    "llvm/bin/clang", "bin/clang-ocl", "llvm/bin/clang++", "llvm/bin/clang-offload-bundler"};
```

Comment:
Don't use `std::string` or `const char*` to represent paths - the forward slash separator does not work on Windows (in most cases). Please use `fs:path` from `miopen\filesystem.hpp` instead.

Reply:
Apart from above, the HIP SDK directory structure is different on Windows.

Reply:
- `<HIP_SDK_Install_Dir>\bin\clang.exe`
- `<HIP_SDK_Install_Dir>\bin\clang++.exe`
- `<HIP_SDK_Install_Dir>\bin\clang-offload-bundler.exe`
There is no `clang-ocl.exe`, regular `clang.exe` does the same. IMPORTANT: extensions are required!

Reply:
@apwojcik Perhaps you ought to take a step back and see what the code is trying to fix before pointing out all the issues with windows? Adding extensions here doesn't help anyone. The code is changing from hard coded #defines (with unix semantics) to things looked up at runtime. It is all under hash defines so if you don't want it on windows (which is reasonable) then you don't have to have it.

Reply:
@apwojcik Can we assume that this enhancement is targeted for Linux only for now? And then extend it to cover Windows later?

Reply:
@amd-isparry, I need you to be aware of how your changes may impact Windows (even if you claim they won't). Otherwise, that is extra work we must take care of with limited resources. We need to be extra careful because we are still adding fixes and changes to the top of the living development branch. Otherwise, we will not be able to meet the deadlines. There is no stable base for Windows yet. I wanted you to know why I am so picky - your PR may mess a lot.

Reply:
Have you compiled your branch on Windows?

Reply:
@apwojcik 
> I need you to be aware of how your changes may impact Windows (even if you claim they won't)

👍